stores/couchbase/src/vector/index.ts

```diff
@@ -0,0 +1,282 @@
+import { MastraVector } from '@mastra/core/vector';
+import type {
+    QueryResult,
+    IndexStats,
+    CreateIndexParams,
+    UpsertVectorParams,
+    QueryVectorParams,
+    ParamsToArgs,
+} from '@mastra/core/vector';
+import { Bucket, Cluster, Collection, connect, SearchRequest, VectorQuery, VectorSearch, Scope } from 'couchbase';
+
+const metricsInMastra = ['cosine', 'euclidean', 'dotproduct'] as const;
+type MastraMetric = typeof metricsInMastra[number];
+const metricsInCouchbase = ['cosine', 'l2_norm', 'dot_product'] as const;
+type CouchbaseMetric = typeof metricsInCouchbase[number];
```

Comment:
Could we make this just the types themselves like this?
```suggestion
type MastraMetric = 'cosine' | 'euclidean' | 'dotproduct';
type CouchbaseMetric = 'cosine' | 'l2_norm' | 'dot_product';
```